graviti/utility/file.py

```diff
@@ -84,14 +113,37 @@ class File(FileBase):
 
     """
 
+    __slots__ = FileBase.__slots__
+
     _BUFFER_SIZE = 65536
 
     def __init__(self, path: str) -> None:
-        self.path = Path(path)
+        self.path = Path(path).absolute()
 
     def _repr_head(self) -> str:
         return f'{self.__class__.__name__}("{self.path.name}")'
 
+    @classmethod
+    def _create(cls: Type[_T], url: str, checksum: str) -> _T:
+        obj: _T = object.__new__(cls)
+        obj.path = Path(url[7:])
+        obj._url = url
```

Comment:
We need add `_url` in `to_pyobj`， We need to save the `path` and `checksum` in pyarrow array.